Sources/TensorFlow/Operators/Image.swift

```diff
@@ -343,3 +343,159 @@ func _vjpResizeBicubic<Scalar: TensorFlowFloatingPoint>(
     }
   )
 }
+
+/// Returns a 2-D dilation with the specified input, filter, strides, and padding.
+///
+/// - Parameters:
+///   - input: The input.
+///   - filter: The dilation filter.
+///   - strides: The strides of the sliding filter for each dimension of the input.
+///   - padding: The padding for the operation
+///   - rates: The dilation rates for each dimension of the input.
+/// - Precondition: `input` must have rank `4`.
+/// - Precondition: `filter` must have rank 4.
+@differentiable(wrt: (input, filter))
+public func dilation2D<Scalar: TensorFlowFloatingPoint>(
+  _ input: Tensor<Scalar>,
+  filter: Tensor<Scalar>,
+  strides: (Int, Int, Int, Int) = (1, 1, 1, 1),
+  rates: (Int, Int, Int, Int) = (1, 1, 1, 1),
+  padding: Padding = .valid
+) -> Tensor<Scalar> {
+  precondition(input.shape.rank == 4, "The input must have rank 4.")
+  precondition(filter.shape.rank == 4, "The filter must have rank 4.")
+  return _Raw.dilation2D(
+    input,
+    filter: filter,
+    strides: [Int32(strides.0), Int32(strides.1), Int32(strides.2), Int32(strides.3)],
+    rates: [Int32(rates.0), Int32(rates.1), Int32(rates.2), Int32(rates.3)],
+    padding: padding.raw
+  )
+}
+
+@usableFromInline
+@derivative(of: dilation2D)
+func _vjpDilation2D<Scalar: TensorFlowFloatingPoint>(
+  _ input: Tensor<Scalar>,
+  filter: Tensor<Scalar>,
+  strides: (Int, Int, Int, Int),
+  rates: (Int, Int, Int, Int),
+  padding: Padding
+) -> (value: Tensor<Scalar>, pullback: (Tensor<Scalar>) -> (Tensor<Scalar>, Tensor<Scalar>)) {
+  let value = dilation2D(input, filter: filter, strides: strides, rates: rates, padding: padding)
+  return (
+    value,
+    { v in
+      (
+        dilation2DBackpropInput(
+          v, input: input, filter: filter,
+          strides: strides, rates: rates, padding: padding),
+        dilation2DBackpropFilter(
+          v, input: input, filter: filter,
+          strides: strides, rates: rates, padding: padding)
+      )
+    }
+  )
+}
+
+/// TensorFlow builtin dilation2d gradient helper for the input.
+@differentiable(wrt: (x, filter))
+@usableFromInline
+func dilation2DBackpropInput<Scalar: TensorFlowFloatingPoint>(
+  _ x: Tensor<Scalar>,
+  input: Tensor<Scalar>,
+  filter: Tensor<Scalar>,
+  strides: (Int, Int, Int, Int) = (1, 1, 1, 1),
+  rates: (Int, Int, Int, Int) = (1, 1, 1, 1),
+  padding: Padding = .valid
+) -> Tensor<Scalar> {
+    return _Raw.dilation2DBackpropInput(
+    input,
+    filter: filter,
+    outBackprop: x,
+    strides: [Int32(strides.0), Int32(strides.1), Int32(strides.2), Int32(strides.3)],
+    rates: [Int32(rates.0), Int32(rates.1), Int32(rates.2), Int32(rates.3)],
+    padding: padding.raw)
+}
+
+@usableFromInline
+@derivative(of: dilation2DBackpropInput)
```

Comment:
```suggestion
@derivative(of: dilation2DBackpropInput, wrt: (x, input))
```

I think this might fix your compile error. (I was confused for a bit too, so I [played around in a colab to confirm](https://colab.research.google.com/drive/1QnyZmpE9O3xGc5uxmLySxpB6jcczQdB2). Happy to provide more explanation if that's helpful!)

Reply:
That did it, thanks!